library/modules/EventManager.cpp

```diff
@@ -1254,3 +1261,32 @@ static void manageInteractionEvent(color_ostream& out) {
         //TODO: deduce attacker from latest defend event first
     }
 }
+
+static void manageActionEvent(color_ostream& out) {
+    if (!df::global::world)
+        return;
+    multimap<Plugin*,EventHandler> copy(handlers[EventType::UNIT_ACTION].begin(), handlers[EventType::UNIT_ACTION].end());
+    for ( size_t a = 0; a < df::global::world->units.all.size(); a++ ) {
+        df::unit* unit = df::global::world->units.all[a];
+        if ( Units::isActive(unit) ) {
```

Comment:
To reduce indentation, how about
```
if (!Units::isActive(unit)) {
    unitToKnownActions.erase(unit->id);
    continue;
}
```
This way you don't need an else clause

Reply:
Nah, I generally prefer things be structured in the way mentioned for stuff like this, I just sort of forget to do it